### PR TITLE
MVP Features and Refactors

### PR DESCRIPTION
**Pull request by dmorejon**
This PR contains many small changes to improve the overall tool.
There are some refactoring changes as well. _transform.js_ is now _transformJS.js_ and the `transformFile()` function in _index.js_ is now in its own file: _transformFile.js_.

- Only .HTML files are transformed, and the tool immediately returns if the file is not HTML.
- Instead of stripping `debug()` calls, we change that to `console.log()` now since they likely contain important context for some tests.
- We remove `testRunner.dumpAsText()` calls now since _testharnessreport.js_ does that by default.
- Use npm debug to log within each module
- A new check to make sure scripts end in newline
- Regex to make sure js-test.js is src'd before transformation, throws error if transformation proceeds but the src was not changed.
- Babel was removing all newlines in the scripts during transformation, so adding the `retain_lines: true` option keeps original newlines, but required some other stylistic tradeoffs like objects being compressed and too many newlines if removals happen. Both scenarios aren't ideal, but I thought it would be better to have too many newlines in logical places than a wall of code for every script.

### CODE REVIEW

File: src/transformJS.js [hunk 1]
```diff
@@ -163,12 +166,31 @@ function removeDescriptionFactory(transformInfo) {
...
-// TODO: maybe change debug() to console.log() instead of deleting?
-function removeDebug() {
+function transformDebug() {
   return {
     visitor: {
       CallExpression(path) {
+        const consoleIdentifier = babel.types.identifier('console');
+        const logIdentifier = babel.types.identifier('log');
+        const consoleLogNode =
+          babel.types.memberExpression(consoleIdentifier, logIdentifier);
```
Comment (Hexcles) [inline src/transformJS.js]: Nit: maybe only construct the expression if the condition is met?

File: test/testTransformFile.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+describe('#transformFile', function() {
+  let tmpDir;
+  before(function() {
+    tmpDir = tmp.dirSync({tmpdir: '.', unsafeCleanup: true});
+  });
+
+  after(function() {
+    tmpDir.removeCallback();
+  });
```
Comment (Hexcles) [inline test/testTransformFile.js:17]: Nit
```suggestion
  });

```

File: src/transformJS.js [hunk 3]
```diff
@@ -4,7 +4,10 @@ const babel = require('@babel/core');
 // addSetupNode adds the setup() call to the beginning of the script.
```
Comment (Hexcles) [inline src/transformJS.js]: Maybe call this file `transformScript.js` for consistency?

File: test/testdata/reference/transformed_canvas-description-and-role.html [hunk 4]
```diff
@@ -17,18 +17,24 @@
 
 <div id="console"></div>
 <script>
+setup({ single_test: true, explicit_done: false });
```
Comment (Hexcles) [inline test/testdata/reference/transformed_canvas-description-and-role.html:20]: I see this is a side-effect of retainLines. I think this is acceptable.

**Review by Hexcles — APPROVED** [review]
LGTM % nits

File: src/transformJS.js (hunk 3 above)
Comment (dmorejon) [reply src/transformJS.js]: Yes! Such a good name. I was not happy with the one I picked

**Pull request merged**